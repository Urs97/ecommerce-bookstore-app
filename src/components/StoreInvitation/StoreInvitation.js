import React from 'react';
import './StoreInvitation.scss';
import photo_sm from '../../assets/images/store-invitation-photo-sm.jpg';
import photo_md from '../../assets/images/store-invitation-photo-md.jpg';
import photo_lg from '../../assets/images/store-invitation-photo-lg.jpg';
import { Link } from 'react-router-dom';

const StoreInvitation = () => {
    return (
        <section className="store-invitation">
            <img src={photo_sm}
                srcSet={`${photo_sm} 640w,
                        ${photo_md} 1024w,
                        ${photo_lg} 2400w,`} 
                alt="Beautiful library interior by Benjamin Ashton on Unsplash"/>
            <h1 className="title">Browse Through Our Complete Library</h1>
            <Link to="/store" className="button">View More</Link>
        </section>
    )
};

export default StoreInvitation;
