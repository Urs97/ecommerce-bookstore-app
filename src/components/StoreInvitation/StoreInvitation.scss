.store-invitation {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    position: relative;
    width: 100%;
    height: 25em;

    & h1 {
        font-size: 2.5em;
        color: #FFFFFF;
        margin-bottom: 1em;
    }

    & img {
        position: absolute;
        object-fit: cover;
        width: 100%;
        height: 25em;
        z-index: -2;
    }

    & .button:focus {
        outline: 0.2em solid #FFFFFF;
    }

    & .button:not(.focus-visible) {
        outline: none;
      }
}

.store-invitation::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -1;
    background-color: rgba(0, 0, 0, 0.5);
}