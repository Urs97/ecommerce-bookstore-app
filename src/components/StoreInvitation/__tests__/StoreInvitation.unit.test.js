import React from "react";
import { shallow } from "enzyme";
import StoreInvitation from "../StoreInvitation";

describe("StoreInvitation", () => {
  test("renders StoreInvitation", () => {
    const wrapper = shallow(<StoreInvitation />);
    expect(wrapper).toMatchSnapshot();
  });
});
