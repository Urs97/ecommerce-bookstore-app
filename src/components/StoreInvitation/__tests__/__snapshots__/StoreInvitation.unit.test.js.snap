// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreInvitation renders StoreInvitation 1`] = `
<section
  className="store-invitation"
>
  <picture>
    <source
      media="(orientation: portrait)"
      srcSet="store-invitation-photo-768-portrait.webp 320w"
      type="image/webp"
    />
    <source
      media="(orientation: portrait)"
      srcSet="store-invitation-photo-768-portrait.jpg 320w"
      type="image/jpg"
    />
    <source
      srcSet="store-invitation-photo-768.webp 320w,
                        store-invitation-photo-1024.webp 1024w,
                        store-invitation-photo-1366.webp 1366w,
                        store-invitation-photo-1600.webp 1600w,
                        store-invitation-photo-1920.webp 1920w"
      type="image/webp"
    />
    <source
      srcSet="store-invitation-photo-768.jpg 320w,
                        store-invitation-photo-1024.jpg 1024w,
                        store-invitation-photo-1366.jpg 1366w,
                        store-invitation-photo-1600.jpg 1600w,
                        store-invitation-photo-1920.jpg 1920w"
      type="image/jpg"
    />
    <img
      alt="Beautiful library interior by Benjamin Ashton/Stephanie Leblanc on Unsplash"
      src="store-invitation-photo-1920.jpg"
    />
  </picture>
  <h1
    className="title"
  >
    Browse Through Our Complete Library
  </h1>
  <Link
    className="button"
    to="/store"
  >
    View More
  </Link>
</section>
`;
